### PR TITLE
WIP Refactor to enable DDF-5802

### PR DESCRIPTION
**Pull request by lamhuy**
#### What does this PR do?

Minor Precursor to address improvement in
https://github.com/codice/ddf/issues/5802

1. Rename SolrCatalogProvider to BaseSolrCatalogProvider
2. Adding timeout for isAvailable check
3. SolrCloudClientFactory handling of multiple collection configs

#### Select relevant component teams: 
@codice/core-apis 
@codice/solr 
#### Ask 2 committers to review/merge the PR and tag them here.

@pklinef 
@bdeining 
@millerw8 

#### What are the relevant tickets?
Fixes: #____

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ x ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by lamhuy** [comment]
build now

File: platform/solr/solr-factory/src/main/java/org/codice/solr/factory/SolrClientFactory.java [hunk 1]
```diff
@@ -30,5 +30,16 @@
    * @return the newly created {@code SolrClient}
    * @throws IllegalArgumentException if <code>core</code> is <code>null</code>
    */
-  public SolrClient newClient(String core);
+  SolrClient newClient(String core);
+
+  boolean isAvailable();
```
Comment (paouelle) [inline platform/solr/solr-factory/src/main/java/org/codice/solr/factory/SolrClientFactory.java]: ✏️ Should add javadoc to this method

File: catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/BaseSolrCatalogProvider.java [hunk 2]
```diff
@@ -163,6 +185,15 @@ public boolean isAvailable(SourceMonitor callback) {
     return isAvailable(); // then trigger an active ping
   }
 
+  public boolean isAvailable(long timeout, long pollInterval, TimeUnit unit) {
+    try {
+      return solr.isAvailable(timeout, pollInterval, unit);
+    } catch (InterruptedException e) {
```
Comment (paouelle) [inline catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/BaseSolrCatalogProvider.java]: ❗️ Improper way to handle interruptions. They need to be propagated up by re-interrupting the current thread

File: platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java [hunk 3]
```diff
@@ -136,7 +139,41 @@ private static boolean solrCoreExists(SolrClient client, String coreName)
...
-    return new SolrClientAdapter(core, () -> createSolrHttpClient(solrUrl, core, coreUrl));
+
+    SolrClientAdapter adaptor =
+        new SolrClientAdapter(core, () -> createSolrHttpClient(solrUrl, core, coreUrl));
+    try {
+      if (!adaptor.isAvailable(30, 5, TimeUnit.SECONDS)) {
+        LOGGER.warn("Solr Client {} is not available after 30 seconds", core);
+      }
+    } catch (InterruptedException e) {
+      LOGGER.error("Unable to connect to solr client {}: {} ", core, e.getStackTrace());
```
Comment (paouelle) [inline platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java]: ❗️ Improper way to handle interruptions. They need to be propagated up by re-interrupting the current thread.

File: platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java [hunk 4]
```diff
@@ -63,18 +64,69 @@
...
         "Solr({}): Creating a Solr Cloud client using Zookeeper hosts [{}]", core, zookeeperHosts);
-    return new SolrClientAdapter(core, () -> createSolrCloudClient(zookeeperHosts, core));
+    SolrClientAdapter adaptor =
+        new SolrClientAdapter(core, () -> createSolrCloudClient(zookeeperHosts, core));
+    try {
+      if (!adaptor.isAvailable(30, 5, TimeUnit.SECONDS)) {
+        LOGGER.warn("Solr Client {} is not available after 30 seconds", core);
+      }
+    } catch (InterruptedException e) {
+      LOGGER.error("Unable to connect to solr client {}: {} ", core, e.getStackTrace());
```
Comment (paouelle) [inline platform/solr/solr-factory-impl/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java]: ❗️ Improper way to handle interruptions. They need to be propagated up by re-interrupting the current thread.

**Comment by lamhuy** [comment]
build now

**Comment by lamhuy** [comment]
closing due to https://github.com/codice/ddf/pull/5813

**Comment by lamhuy** [comment]
closing as this PR is eclipsed by https://github.com/codice/ddf/pull/5813

**Pull request closed without merging**